### PR TITLE
Concourse update

### PR DESCRIPTION
**Pull request by cerdmann-pivotal**
### Best Practice Checklist
Check these items before submitting a PR... See the Contribution Guide for the full detail: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md 

1. uri's compatible with all clouds (Stack, China, Government)
1. Staged artifacts use _artifactsLocation & _artifactsLocationSasToken
1. Use resourceGroup().location for resource locations
1. Folder names for artifacts (nestedtemplates, scripts, DSC)
1. Use literal values for apiVersion (no variables)
1. Parameter files (GEN-UNIQUE for value generation and no "changemeplease" values
1. $schema and other uris use https
1. Use uniqueString() whenever possible to generate names for resources.  While this is not required, it's one of the most common failure points in a deployment. 
1. Update the metadata.json with the current date

For details: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/bp-checklist.md

- [X] - Please check this box once you've submitted the PR if you've read through the Contribution Guide and best practices checklist.

### Changelog

* Updated bosh release, stemcells, azure cpi to latest version
* Updated concourse release and garden-runc
* Previous 1.0.2 no longer worked. Used azure-quickstart-templates/bosh setup as template for changes to scripts and manifests
* Added randomized passwords in bosh.yml

**Please note that [this line](https://github.com/cerdmann-pivotal/azure-quickstart-templates/blob/concourse-update/concourse-ci/azuredeploy.json#L116) needed to be changed for testing to point to the assets in my branch. This line should not be merged, but should stay the same as [this line](https://github.com/Azure/azure-quickstart-templates/blob/master/concourse-ci/azuredeploy.json#L116). If you do not merge this line for testing, you will pull the current scripts and manifests from master and you will not see the results of this change. I was unsure of how to check this in to test the change.**

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@cerdmann-pivotal__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <span>
        In order for us to evaluate and accept your PR, we ask that you sign a contribution license agreement. It's all electronic and will take just minutes. I promise there's no faxing. https://cla.azure.com.
    </span>

TTYL, AZPRBOT;

**Comment by cerdmann-pivotal** [comment]
Working on the CLA

**Comment by azurecla** [comment]
__@cerdmann-pivotal__, Thanks for signing the contribution license agreement so quickly! Actual humans will now validate the agreement and then evaluate the PR.
<br />Thanks, AZPRBOT;

File: concourse-ci/azuredeploy.json [hunk 1]
```diff
@@ -113,7 +113,7 @@
     "ubuntuOSVersion": "14.04.3-LTS",
     "sshKeyPath": "[concat('/home/', parameters('adminUsername'), '/.ssh/authorized_keys')]",
 
-    "baseUriAzureCloud": "https://raw.githubusercontent.com/Azure/azure-quickstart-templates/master/concourse-ci/",
+    "baseUriAzureCloud": "https://raw.githubusercontent.com/cerdmann-pivotal/azure-quickstart-templates/concourse-update/concourse-ci/",
```
Comment (bingosummer) [inline concourse-ci/azuredeploy.json]: @cerdmann-pivotal 
The best practice by azure-quickstart-templates is to add a parameter `_artifactsLocation`. For testing, you can specify the parameter with your private URL.
FYI:
https://github.com/Azure/azure-quickstart-templates/blob/master/bosh-setup/azuredeploy.json#L88-L95

File: concourse-ci/README.md [hunk 2]
```diff
@@ -1,9 +1,9 @@
 # Setup Concourse CI with Bosh
 
-<a href="https://portal.azure.com/#create/Microsoft.Template/uri/https%3A%2F%2Fraw.githubusercontent.com%2FAzure%2Fazure-quickstart-templates%2Fmaster%2Fconcourse-ci%2Fazuredeploy.json" target="_blank">
+<a href="https://portal.azure.com/#create/Microsoft.Template/uri/https%3A%2F%2Fraw.githubusercontent.com%2Fcerdmann-pivotal%2Fazure-quickstart-templates%2Fconcourse-update%2Fconcourse-ci%2Fazuredeploy.json" target="_blank">
     <img src="http://azuredeploy.net/deploybutton.png"/>
 </a>
-<a href="http://armviz.io/#/?load=https%3A%2F%2Fraw.githubusercontent.com%2FAzure%2Fazure-quickstart-templates%2Fmaster%2Fconcourse-ci%2Fazuredeploy.json" target="_blank">
+<a href="http://armviz.io/#/?load=https%3A%2F%2Fraw.githubusercontent.com%2Fcerdmann-pivotal%2Fazure-quickstart-templates%2Fconcourse-update%2Fconcourse-ci%2Fazuredeploy.json" target="_blank">
     <img src="http://armviz.io/visualizebutton.png"/>
 </a>
```
Comment (bingosummer) [inline concourse-ci/README.md]: @cerdmann-pivotal 
I understand that you changed the link temporally for testing. But it should not be in the PR.

**Comment by bingosummer** [comment]
@cerdmann-pivotal 
Thanks for helping updating this template.
I am happy to have a review and leave some comments although I don't have the privilege to merge the PR.

File: concourse-ci/scripts/setup_env [hunk 3]
```diff
@@ -37,15 +50,50 @@ function get_setting() {
...
+  sudo mv ./bosh-init-* /usr/local/bin/bosh-init
+}
+
 environment=$(get_setting ENVIRONMENT)
 
 set +e
 
 echo "Start to install python packages..."
-pkg_list="msrest msrestazure azure==2.0.0rc1 azure-storage netaddr"
+pkg_list="msrest==0.4.4 msrestazure==0.4.4 requests==2.11.1 azure==2.0.0rc1 netaddr==0.7.18"
```
Comment (bingosummer) [inline concourse-ci/scripts/setup_env]: Please add pip==1.5.4 setuptools==32.3.1.
FYI:
https://github.com/Azure/azure-quickstart-templates/commit/d6e3e90126fa0a5dacec1cb00d79622624345e80

Comment (cerdmann-pivotal) [inline concourse-ci/scripts/setup_env]: Ok, let me make these changes

**Comment by cerdmann-pivotal** [comment]
Changes are made

**Comment by bingosummer** [comment]
Thanks @cerdmann-pivotal .
LGTM.
@gatneil Could you please have a look?

File: concourse-ci/README.md [hunk 4]
```diff
@@ -1,9 +1,9 @@
 # Setup Concourse CI with Bosh
 
-<a href="https://portal.azure.com/#create/Microsoft.Template/uri/https%3A%2F%2Fraw.githubusercontent.com%2FAzure%2Fazure-quickstart-templates%2Fmaster%2Fconcourse-ci%2Fazuredeploy.json" target="_blank">
+<a href="https://portal.azure.com/#create/Microsoft.Template/uri/https%3A%2F%2Fraw.githubusercontent.com%2FAzure%2Fazure-quickstart-templates%2Fconcourse-update%2Fconcourse-ci%2Fazuredeploy.json" target="_blank">
```
Comment (gatneil) [inline concourse-ci/README.md]: shouldn't this point to master, not concourse-update?

File: concourse-ci/README.md [hunk 5]
```diff
@@ -1,9 +1,9 @@
...
     <img src="http://azuredeploy.net/deploybutton.png"/>
 </a>
-<a href="http://armviz.io/#/?load=https%3A%2F%2Fraw.githubusercontent.com%2FAzure%2Fazure-quickstart-templates%2Fmaster%2Fconcourse-ci%2Fazuredeploy.json" target="_blank">
+<a href="http://armviz.io/#/?load=https%3A%2F%2Fraw.githubusercontent.com%2Fcerdmann-pivotal%2Fazure-quickstart-templates%2Fconcourse-update%2Fconcourse-ci%2Fazuredeploy.json" target="_blank">
```
Comment (gatneil) [inline concourse-ci/README.md]: Shouldn't this point to Azure master?

File: concourse-ci/manifests/bosh.yml [hunk 6]
```diff
@@ -16,24 +16,25 @@ resource_pools:
...
+- name: public
+  type: vip
+
 - name: private
   type: manual
   subnets:
   - range: REPLACE_WITH_SUBNET_ADDRESS_RANGE_FOR_BOSH
     gateway: REPLACE_WITH_GATEWAY_IP
-    dns: [168.63.129.16]
+    dns: [168.63.129.16, 8.8.8.8]
```
Comment (gatneil) [inline concourse-ci/manifests/bosh.yml]: Interesting; why both Azure and Google DNS? Not really a problem, just curious

**Comment by gatneil** [comment]
Just a couple comments; otherwise looks good :)

File: concourse-ci/manifests/bosh.yml (hunk 6 above)
Comment (bingosummer) [inline concourse-ci/manifests/bosh.yml]: `8.8.8.8` is used as the second DNS in case Azure DNS doesn't work.

File: concourse-ci/README.md (hunk 4 above)
Comment (bingosummer) [inline concourse-ci/README.md]: I think so.
@cerdmann-pivotal

File: concourse-ci/README.md (hunk 5 above)
Comment (bingosummer) [inline concourse-ci/README.md]: I think so.

**Comment by bingosummer** [comment]
Thanks Neil.

File: concourse-ci/README.md (hunk 5 above)
Comment (cerdmann-pivotal) [inline concourse-ci/README.md]: Changes are made

**Comment by gatneil** [comment]
Looks good; merging :)

**Pull request merged**